apis/apiextensions/v1/xrd_webhooks.go

```diff
@@ -26,17 +27,21 @@ import (
 const (
 	errUnexpectedType = "unexpected type"
 
-	errGroupImmutable       = "spec.group is immutable"
-	errPluralImmutable      = "spec.names.plural is immutable"
-	errKindImmutable        = "spec.names.kind is immutable"
-	errClaimPluralImmutable = "spec.claimNames.plural is immutable"
-	errClaimKindImmutable   = "spec.claimNames.kind is immutable"
+	errGroupImmutable                  = "spec.group is immutable"
+	errPluralImmutable                 = "spec.names.plural is immutable"
+	errKindImmutable                   = "spec.names.kind is immutable"
+	errClaimPluralImmutable            = "spec.claimNames.plural is immutable"
+	errClaimKindImmutable              = "spec.claimNames.kind is immutable"
+	errConversionWebhookConfigRequired = "spec.conversion.webhook is required when spec.conversion.strategy is 'Webhook'"
 )
 
 // +kubebuilder:webhook:verbs=update,path=/validate-apiextensions-crossplane-io-v1-compositeresourcedefinition,mutating=false,failurePolicy=fail,groups=apiextensions.crossplane.io,resources=compositeresourcedefinitions,versions=v1,name=compositeresourcedefinitions.apiextensions.crossplane.io,sideEffects=None,admissionReviewVersions=v1
 
 // ValidateCreate is run for creation actions.
 func (in *CompositeResourceDefinition) ValidateCreate() error {
+	if conversion := in.Spec.Conversion; conversion != nil && conversion.Strategy == extv1.WebhookConverter && conversion.Webhook == nil {
```

Comment:
```suggestion
	if c := in.Spec.Conversion; c != nil && c.Strategy == extv1.WebhookConverter && c.Webhook == nil {
```

Nit: I think a short variable is plenty here given that `in.Spec.Conversion` appears in the same line.

https://github.com/crossplane/crossplane/tree/master/contributing#use-descriptive-variable-names-sparingly